homeassistant/components/trend/binary_sensor.py

```diff
@@ -99,42 +101,40 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the trend sensors."""
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
-
-    sensors = []
-
-    for device_id, device_config in config[CONF_SENSORS].items():
-        entity_id = device_config[ATTR_ENTITY_ID]
-        attribute = device_config.get(CONF_ATTRIBUTE)
-        device_class = device_config.get(CONF_DEVICE_CLASS)
-        friendly_name = device_config.get(ATTR_FRIENDLY_NAME, device_id)
-        invert = device_config[CONF_INVERT]
-        max_samples = device_config[CONF_MAX_SAMPLES]
-        min_gradient = device_config[CONF_MIN_GRADIENT]
-        sample_duration = device_config[CONF_SAMPLE_DURATION]
-        min_samples = device_config[CONF_MIN_SAMPLES]
-
-        sensors.append(
-            SensorTrend(
-                hass,
-                device_id,
-                friendly_name,
-                entity_id,
-                attribute,
-                device_class,
-                invert,
-                max_samples,
-                min_gradient,
-                sample_duration,
-                min_samples,
+
+    for sensor_name, sensor_config in config[CONF_SENSORS].items():
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={CONF_NAME: sensor_name, **sensor_config},
```

Comment:
For helpers we keep the YAML support normally, so I suggest we do it here too.

Reply:
Re-added the YAML support